### Eclipse Workspace Patch 1.0
#P jmesa-examples
Index: web/jsp/portlet.jsp
===================================================================
--- web/jsp/portlet.jsp	(revision 2031)
+++ web/jsp/portlet.jsp	(working copy)
@@ -10,31 +10,30 @@
 	<div><c:out value="${table.markup}" escapeXml="false" /></div>
 </form>
 
-<script type="text/javascript">
+<script type="text/javascript">
+$(function() {
+
+	var url = '<c:url value="/portlet.run" />?id=${table.id}&';
+
+<c:if test="${useAjax}">	
+	$('#${table.id}').invokeAction(function(action) {
+		var pString = $(this).setExportToLimit('').createParameterStringForLimit();
+		$.get(url + pString, function(data) {
+			$('#${table.id}').parents("div:first").html(data);
+		});
+	});
+</c:if>
+
+<c:if test="${!useAjax}">
+	$('#${table.id}').invokeAction(function(action) {
+		$(this).setExportToLimit('').createHiddenInputFieldsForLimitAndSubmit();
+    });
+</c:if>
+	
+	$('#${table.id}').invokeExportAction(function() {
+		location.href = url + $(this).createParameterStringForLimit();
+	});
+	
+});
 
-	var url = '<c:url value="/portlet.run" />?id=${table.id}&';
-
-<c:if test="${useAjax}">
-	function onInvokeAction(id) {
-		setExportToLimit(id, '');
-		var parameterString = createParameterStringForLimit(id);
-		$.get(url + parameterString, function(data) {
-			$('#${table.id}').parents("div:first").html(data);
-		});
-	}
-</c:if>
-
-<c:if test="${!useAjax}">
-	function onInvokeAction(id, action) {
-		setExportToLimit(id, '');
-		createHiddenInputFieldsForLimitAndSubmit(id);
-    }
-</c:if>
-
-	function onInvokeExportAction(id) {
-		var parameterString = createParameterStringForLimit(id);
-		console.log(TableFacadeManager.getTableFacade(id).limit);
-		location.href = url + parameterString;
-	}
-
 </script>
\ No newline at end of file
